addon/mixins/model-records/relationship-route-mixin.js

```diff
@@ -1,26 +1,34 @@
 import Ember from 'ember'
 
 const {
-  Mixin,
-  String: { singularize }
+  Mixin
 } = Ember
 
 export default Mixin.create({
   setupController(controller, model) {
     this._super(controller, model)
 
     controller.setProperties({
-      parentModel: this.modelFor('admin.model-records.show'),
-      recordName: this._relationshipName(),
+      parentModel: this._parentModel(),
+      relationshipName: this._relationshipName(),
       recordType: this._relationshipType()
     })
   },
 
+  _parentModel() {
+    return this.modelFor('admin.model-records.show')
```

Comment:
this seems to assume that modelFor is defined.  Is this only being mixed in to one type of object?

Reply:
Yeah, this should only be mixed into routes